### PR TITLE
Jenkins integration

### PR DESCRIPTION
**Pull request by antongulenko**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    stages {
+        stage('Build & test') { 
+            steps {
+                    sh 'go clean -i -v ./...'
+                    sh 'go install -v ./...'
+                    sh 'rm -rf reports && mkdir -p reports'
+                    sh 'go test -v ./... 2>&1 | go-junit-report > reports/test.xml'
+                    sh 'go vet ./... &> reports/vet.txt'
+                    sh 'golint $(go list -f "{{.Dir}}" ./...) &> reports/lint.txt'
```
Comment (srnbckr) [inline Jenkinsfile:16]: Do you want to use the lint report somewhere?

**Comment by antongulenko** [comment]
Yes, the SonarQube Golang Scanner has the ability to parse the go vet and
golint outputs :) I prepared them for it, but for now they are just
archived. I don't know yet how test coverage works.

File: Jenkinsfile (hunk 1 above)
Comment (antongulenko) [reply Jenkinsfile:16]: Yes, the SonarQube Golang Scanner has the ability to parse the go vet and
golint outputs :) I prepared them for it, but for now they are just
archived. I don't know yet how test coverage works.

Comment (antongulenko) [reply Jenkinsfile:16]: BTW, I found that runnin each command in its individual 'sh' is very convenient in the BlueOcean view, as the output of every command is then captured and displayed individually.

**Comment by srnbckr** [comment]
great! I will merge it now and add the sonarqube scanner in another PR.

**Review by srnbckr — APPROVED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (srnbckr) [reply Jenkinsfile:16]: True, maybe I will change the other Jenkinsfile as well

**Pull request merged**